### PR TITLE
Add config schema

### PR DESCRIPTION
**Pull request by dhruvin2910**

### CODE REVIEW

**Comment by dhruvin2910** [comment]
This is initial config schema planned for 0.13 release.
Any changes done to the config.go must be reflected here as well.
`additionalProperties: false` is omitted deliberately until the release of 0.13 (as it'll restrict adding new features' configs).

File: protocol/config.schema.json [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+      "properties": {
+        "rule": {
+          "type": "string",
+          "enum": [
+            "allow",
+            "deny",
+            "match",
+            "and",
+            "or",
+            "service",
```
Comment (YourTechBud) [inline protocol/config.schema.json]:
```suggestion
            "webhook",
```

File: protocol/config.schema.json [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+              "col",
+              "find"
+            ]
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "rule": {
+                "const": "service"
```
Comment (dhruvin2910) [inline protocol/config.schema.json]: This must be changed as well, is it?

Comment (YourTechBud) [reply protocol/config.schema.json]: Yes

**Comment by Jayesh333** [comment]
Hey @dhruvin2910, we have made some minor changes in the config structs present in `config/config.go`. Can you update the JSON schema accordingly?

**Comment by dhruvin2910** [comment]
I see `Static` and `Deploy` fields among others in .13 branch. We discussed over a call about removing them.

@Jayesh333 Are those changes reflected or amended?

**Comment by Jayesh333** [comment]
Space Cloud has no support for static and deploy module in `v0.13.0`. @YourTechBud Let's remove the unused things from config.go.

**Comment by YourTechBud** [comment]
Yeah @Jayesh333 and @dhruvin2910. I'll remove the unwanted config

**Comment by YourTechBud** [comment]
@dhruvin2910 I've removed the unwanted config. You can make the changes now.

**Comment by dhruvin2910** [comment]
On it. There's a type on [config.go](https://github.com/spaceuptech/space-cloud/blob/master/config/config.go#L21).

**Comment by dhruvin2910** [comment]
We discussed about the timeout field in EventingRule. It's not there anymore right?

**Comment by YourTechBud** [comment]
Hey @dhruvin2910, can you fix the typo in this PR itself? Timeout isn't implemented yet. Let's leave it for now

**Pull request merged**